instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/tracers/graphql_tracer.rb

```diff
@@ -27,8 +27,20 @@ class GraphQLTracer < ::GraphQL::Tracing::PlatformTracing
           def platform_trace(platform_key, key, data)
             return yield if platform_key.nil?
 
-            tracer.in_span(platform_key, attributes: attributes_for(key, data)) do |_span|
-              yield
+            tracer.in_span(platform_key, attributes: attributes_for(key, data)) do |span|
+               yield.tap do |response|
+                if key == 'validate' && !response[:errors]&.empty?
+                  span.status = OpenTelemetry::Trace::Status.error
+                  response[:errors].each do |error|
+                    span.add_event(
+                      "graphql error",
+                      attributes: {
+                        "error" => error.to_h.to_json
```

Comment:
error.to_h might be a nested hash, https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/common.md#attributes. To make sure we include everything, maybe report it as json string?